File: src/Survey/MainBundle/Entity/Question.php
<?php
namespace Survey\MainBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use DomainException;

/**
 * @ORM\Entity
 * @ORM\Table(name="question")
 */
class Question
{
    private $category_mappings = array(
        "adm" => "Administration (Leadership)",
        "apo" => "Apostleship (Missionary)",
        "cou" => "Counsellor (Exhorter)",
        "dis" => "Discernment",
        "eva" => "Evangelist",
        "fai" => "Faith",
        "giv" => "Giving",
        "hea" => "Healing",
        "int" => "Interpretation of Tongues",
        "kno" => "Knowledge",
        "mir" => "Miracles",
        "pro" => "Prophecy (Preacher/Proclaimer)",
        "she" => "Sheperding (Pastor/Teacher)",
        "ser" => "Serving (Helps)",
        "sho" => "Showing Mercy (Compassion)",
        "tea" => "Teaching",
        "ton" => "Tongues",
        "wis" => "Wisdom"
    );

    /**
     * @ORM\Column(type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;

    /**
     * @ORM\Column(type="string", length=300)
     */
    protected $text;

    /**
     * @ORM\Column(type="string", length=10)
     */
    protected $category;

    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set text
     *
     * @param string $text
     * @return Question
     */
    public function setText($text)
    {
        $this->text = $text;

        return $this;
    }

    /**
     * Get text
     *
     * @return string 
     */
    public function getText()
    {
        return $this->text;
    }

    /**
     * Set category. If it is not a valid category, DomainException will be
     *  thrown
     *
     * @param string $category
     * @return Question
     * @throws DomainException
     */
    public function setCategory($category)
    {
        if($this->isValidCategory($category))
        {
            $this->category = $category;
        }
        else
        {
            throw new DomainException("Category " . $category . " is not a valid category");
        }

        return $this;
    }

    /**
     * Get category
     *
     * @return string 
     */
    public function getCategory()
    {
        return $this->category;
    }

    /**
     * Get the expanded version of a category, i.e. "hea" becomes "Healing"
     * 
     * @return string
     */
    public function getPrettyCategory()
    {
        return $this->category_mappings[$this->getCategory()];
    }

    /**
     * Checks if the category passed in is valid
     * 
     * @return boolean
     */
    public function isValidCategory($category)
    {
        return array_key_exists($category, $this->category_mappings);
    }
}

File: src/app/models/question.py
"""
Question model
"""
from google.appengine.ext import ndb


class Question(ndb.Model):

    text = ndb.TextProperty()
    category = ndb.StringProperty()
    question_number = ndb.IntegerProperty()

    CATEGORY_MAPPINGS = {
        "adm": "Administration (Leadership)",
        "apo": "Apostleship (Missionary)",
        "cou": "Counsellor (Exhorter)",
        "dis": "Discernment",
        "eva": "Evangelist",
        "fai": "Faith",
        "giv": "Giving",
        "hea": "Healing",
        "int": "Interpretation of Tongues",
        "kno": "Knowledge",
        "mir": "Miracles",
        "pro": "Prophecy (Preacher/Proclaimer)",
        "she": "Sheperding (Pastor/Teacher)",
        "ser": "Serving (Helps)",
        "sho": "Showing Mercy (Compassion)",
        "tea": "Teaching",
        "ton": "Tongues",
        "wis": "Wisdom"
    }

    @classmethod
    def build_key(cls, question_number):
        """
        Build a key for a new Question.
        :param question_number:
        :return:
        """
        return ndb.Key(cls, question_number)

    @classmethod
    def create(cls, text, category, question_number):
        """
        Create a new question.
        :param text:
        :param category:
        :param question_number:
        :return:
        """
        question = Question(
            key=cls.build_key(question_number),
            text=text,
            category=category,
            question_number=question_number
        )
        question.put()

    @property
    def pretty_category(self):
        """ Return the full (pretty) category name. """
        return self.CATEGORY_MAPPINGS[self.category]

    @classmethod
    def get_all_questions(cls, ordered=False, limit=180):
        """ Returns all questions in a random order. """
        if ordered:
            return cls.query().order(cls.question_number).fetch(limit=limit)

        return cls.query().fetch(limit=limit)

    @classmethod
    def get_by_question_number(cls, question_number):
        """
        Get a question by question number (i.e. 3)
        :param question_number:
        :return:
        """
        return cls.build_key(question_number).get()

    @classmethod
    def get_questions_by_number_range(cls, from_number, to_number):
        """ Gets questions by a range of question_numbers (21 - 40). Returned in ascending order. """
        if not from_number:
            raise ValueError('from_number is required')
        if not to_number:
            raise ValueError('to_number is required')

        return cls.query(cls.question_number >= from_number, cls.question_number <= to_number)\
            .order(cls.question_number).fetch()
